### PR TITLE
fix: limit `origin` length

### LINKED ISSUE #2180: Error 1337: {"origin": [Ensure this field has no more than 200 characters]}
Issue Category
Transaction execution issue - https://github.com/5afe/safe-support/issues/133

What happened?
User getting Error 1337: {"origin": [Ensure this field has no more than 200 characters]} after trying sign approval for stETH (defisaver aave v3 receipt:https://app.defisaver.com/recipes/book?recipe=leverage-wsteth-eth-using-aave-v3)
image

Tried the following:

Ledger and Trezor
Ledger and Trezor imported on Metamask
Was not able to test with hot wallet as user did not have one
Safe address:
eth:0x4D540D0A975351Dbfe82b3Af77ac9E52EeCb54f3

Date and time this happened or you first noticed this issue
03-02-2023

Network
Ethereum Mainnet

Safe Address
0x4D540D0A975351Dbfe82b3Af77ac9E52EeCb54f3

Comment from the support ticket : "It now happens again with the Hedgley app with URLs like https://app.hedgey.finance/escrowless-token-swaps/received/ZXRoZXJldW0tdW5kZWZpbmVk"

### PR DESCRIPTION
**Pull request by iamacook**
## What it solves

Resolves #2180

## How this PR fixes it

The `origin` is now limited to a total of 200 characters as follows. The `name` can be as long as necessary as long as it's less than 200 characters and the `url` is takes up the remaining characters

## How to test it

See issue description.

## Checklist
* [ ] I've tested the branch on mobile 📱
* [ ] I've documented how it affects the analytics (if at all) 📊
* [x] I've written a unit/e2e test for it (if applicable) 🧑‍💻

### CODE REVIEW

**Comment by katspaugh** [comment]
It should be the other way round. Prio to the URL, name up to the rest of chars.

**Review by katspaugh — APPROVED** [review]

File: src/utils/__tests__/transactions.test.ts [hunk 1]
```diff
@@ -59,4 +65,34 @@ describe('transactions', () => {
...
+
+      const result = getTxOrigin(app)
+
+      expect(result?.length).toBe(200)
+
+      expect(result).toBe(
+        '{"url":"https://test.com/aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa","name":"Tes"}',
+      )
+    })
+  })
```
Comment (usame-algan) [inline src/utils/__tests__/transactions.test.ts:97]: I would suggest adding another test case where the URL is long enough that it would also be cut-off and name would be empty

Comment (iamacook) [reply src/utils/__tests__/transactions.test.ts:112]: I've added one.

**Review by usame-algan — APPROVED** [review]
